ddtrace/tracer/spancontext.go

```diff
@@ -523,7 +520,16 @@ func (t *trace) finishedOne(s *span) {
 }
 
 func (t *trace) finishChunk(tr *tracer, ch *chunk) {
-	atomic.AddUint32(&tr.spansFinished, uint32(len(ch.spans)))
+	for _, sp := range ch.spans {
+		if sp == nil {
```

Comment:
Can this happen?
The old way of just adding by `len(ch.spans)` makes more sense to me than iterating over the list and incrementing per span. If we're gonna iterate over this whole list again, we might as well just increment spansFinished directly at every call to span.Finish.